src/d/semantic/caster.d

```diff
@@ -333,9 +338,21 @@ struct Caster(bool isExplicit, alias bailoutOverride = null) {
 					return bailout(t);
 				}
 			
-			case Float, Double, Real :
-				assert(0, "Floating point casts are not implemented");
-			
+			case Float, Double, Real:
+				bool needsExplicit = false;
+				if (isFloat(bt))
+				{
+					const fromSize = getSize(t);
+					const toSize = getSize(bt);
+					if (toSize == fromSize)
+						return CastKind.Exact;
+					else if (toSize > fromSize)
+						return CastKind.SPad;
+					else
+						return CastKind.Trunc;
+				} else {
+					return isExplicit ? CastKind.FloatToInteger : CastKind.Invalid;
+				}
```

Comment:
This whole thing is probably not correct. SPad? Trunc?

Reply:
Also, `toSize == fromSize` doesn't looks like it should be possible.

Reply:
I reused the `CastKind`s but adding float specific ones is probably reasonable for float to float operations.

Reply:
`FloatNarrow`, `FloatWiden`? Or even just `FloatToFloat`?

Reply:
Is there a reason to distinguish narrowing and widening of floats? If yes, the the former, if not, then the later. In doubt, go for the former.

Reply:
It looks like LLVM makes the difference. We probably should.

Reply:
I have changed them